netann/chan_status_manager.go

```diff
@@ -627,7 +631,24 @@ func (m *ChanStatusManager) signAndSendNextUpdate(outpoint wire.OutPoint,
 		return err
 	}
 
-	return m.cfg.ApplyChannelUpdate(chanUpdate)
+	_, _, EdgePolicy, err := m.cfg.Graph.FetchChannelEdgesByOutpoint(&outpoint)
+	if err != nil {
+		return err
+	}
+
+	err = m.cfg.ApplyChannelUpdate(chanUpdate)
+	if err != nil {
+		return err
+	}
+
+	// UpdatePolicy must be called after ApplyChannelUpdate for the updates
+	// to be properly propogated.
+	err = m.cfg.UpdatePolicy(routing.ChannelPolicy{
+		TimeLockDelta: uint32(EdgePolicy.TimeLockDelta),
+		MaxHTLC:       EdgePolicy.MaxHTLC,
+		MinHTLC:       &EdgePolicy.MinHTLC,
```

Comment:
Why not include the disabled bit in the main struct for `UpdatePolicy`? Also rather than do two calls here, we can instead nest this _new_ call within the call to `ApplyChannelUpdate` given that's a function closure as is.

Reply:
So if you're seeing duplicate channel updates sent in the `itest`, then this is the culprit. The `ApplyChannelUpdate` method will send the new channel update all the way to the gossiper, then this method will generate another set of updates further down the line: https://github.com/lightningnetwork/lnd/blob/a329c8061266418bccd797aa5d7d277c736ee930/routing/localchans/manager.go#L114

Reply:
What I think we want to do here instead is modify the `ApplyChannelUpdate` closure to also send these updates to the link.